### PR TITLE
style: Cleanup `Interpolate` code.

### PR DESCRIPTION
**Pull request by emilio**
---

- [x] `./mach build -d` does not report any errors
- [x] `./mach test-tidy` does not report any errors

- [x] These changes do not require tests because refactoring.

---

This change is [Reviewable](https://reviewable.io/reviews/servo/servo/11924)

### CODE REVIEW

**Comment by nox** [comment]
Good idea.

@bors-servo r+

File: components/style/properties/helpers/animated_properties.mako.rs [hunk 1]
```diff
@@ -396,30 +361,31 @@ impl Interpolate for LengthOrPercentage {
...
-                Some(LengthOrPercentageOrAuto::Auto)
+                Ok(LengthOrPercentageOrAuto::Auto)
             }
             (this, other) => {
                 let this: Option<CalcLengthOrPercentage> = From::from(this);
                 let other: Option<CalcLengthOrPercentage> = From::from(other);
-                this.interpolate(&other, time).unwrap_or(None).and_then(|value| {
-                    Some(LengthOrPercentageOrAuto::Calc(value))
-                })
+                if let Ok(maybe) = this.interpolate(&other, time) {
```
Comment (Ms2ger) [inline components/style/properties/helpers/animated_properties.mako.rs]:
```rust
if let Ok(Some(result)) = this.interpolate(&other, time) {
```
(though I'd use a `match`)

**Comment by nox** [comment]
Let's take the intermittent failure as an opportunity to address @Ms2ger's nit.

**Comment by emilio** [comment]
Nit addressed. For the record, the intermittent was #11703.

@bors-servo: r=nox,Ms2ger

**Pull request merged**